packages/react/src/React.ml

```diff
@@ -386,13 +386,15 @@ type element =
   | Consumer of element
   | Suspense of { key : string option; children : element; fallback : element }
 
-and client_props = (string * client_prop) list
+and client_props = (string * rsc_value) list
 
-and client_prop =
+and rsc_value =
   (* TODO: Do we need to add more types here? *)
-  | Json : Yojson.Basic.t -> client_prop
-  | Element : element -> client_prop
-  | Promise : 'a Js.Promise.t * ('a -> Yojson.Basic.t) -> client_prop
+  | RSC_value_List : rsc_value list -> rsc_value
+  | RSC_value_Assoc : (string * rsc_value) list -> rsc_value
+  | RSC_value_Json : Yojson.Basic.t -> rsc_value
+  | RSC_value_Element : element -> rsc_value
+  | RSC_value_Promise : 'a Js.Promise.t * ('a -> rsc_value) -> rsc_value
```

Comment:
We might need to change the name from client_prop to `value` or similar. 

What I'm not so sure is the idea to expose List/Assoc in here. Since Json is all we need.

`RSC_value_List xx` is the same as `RSC_value_Json 'List ()`

Also, a worth note about naming and casing. The prefix of `rsc_` is not ideal, and RSC_value into all variants is criminal.

Reply:
> What I'm not so sure is the idea to expose List/Assoc in here. Since Json is all we need.

We cannot use the Json type to compose with other types as Promise, for example.

This is not allowed:
```reasonml
let prop = {
  foo: "Hey",
  bar: {
     element: <div/>
  }
}
```

Reply:
> Also, a worth note about naming and casing. The prefix of rsc_ is not ideal, and RSC_value in all variants is criminal.

I agree. I just pushed with this name because I had no idea, and making and discussing it in this PR would have been better.

The value that I created was:
```
  | List : rsc_value list -> rsc_value
  | Assoc : (string * rsc_value) list -> rsc_value
  | Json : Yojson.Basic.t -> rsc_value
  | Element : element -> rsc_value
  | Promise : 'a Js.Promise.t * ('a -> rsc_value) -> rsc_value
```
But I got the error for the List cause `React.List` already exists, so I pushed this prefix to discuss here.

Reply:
> > What I'm not so sure is the idea to expose List/Assoc in here. Since Json is all we need.
> 
> We cannot use the Json type to compose with other types as Promise, for example.
> 
> This is not allowed:
> 
> ```
> let prop = {
>   foo: "Hey",
>   bar: {
>      element: <div/>
>   }
> }
> ```

Adding `Assoc` or similar would be sufficient to support it, right? No need to add List

Reply:
But then, how would we represent this:
```
let prop = {
  foo: "Hey",
  bar: [<div/>, "Yah!"]
}
```
?
Assoc is: (string, rsc_value) list
List is: rsc_value list

Reply:
```
let props = {
  foo: "Hey",
  bar: [<div/>, "Yah!"]
}

let props = Assoc([("foo", 'String "Hey"), ("bar", 'List(...))])
```

Inside Assoc there're all the Json constructors. That's why I would design a type without even Assoc (and use Json + Promise + Element)